### PR TITLE
EPMRPP-37008: Admin part. Event monitoring. Table

### PR DESCRIPTION
**Pull request by Shyscarecrow**

### CODE REVIEW

File: app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.scss [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  font-size: 12px;
+  @media (max-width: $SCREEN_XS_MAX) {
+    width: 100%;
+  }
+}
+.user-col {
+  width: 15%;
+  vertical-align: top;
+  font-size: 13px;
+  font-weight: bold;
```
Comment (DiscoElevator) [inline app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.scss]: `font-family: $FONT-SEMIBOLD;`

File: app/src/pages/admin/projectEventsPage/projectEventsPage.jsx [hunk 2]
```diff
@@ -68,9 +71,11 @@ export class ProjectEventsPage extends Component {
       onChangePage,
       onChangePageSize,
       loading,
+      events,
     } = this.props;
     return (
       <React.Fragment>
+        <EventsGrid data={events} loading={false} />
```
Comment (DiscoElevator) [inline app/src/pages/admin/projectEventsPage/projectEventsPage.jsx]: `loading={loading}`

File: app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.jsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+UserColumn.propTypes = {
+  className: PropTypes.string.isRequired,
+  value: PropTypes.object,
+};
+UserColumn.defaultProps = {
+  value: {},
+};
+
+const ActionColumn = ({ className, value }) => {
+  const formattedActionType = value.actionType.split(/(?=[A-Z])/).join(' ');
```
Comment (AmsterGet) [inline app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.jsx]: I think we need to match each type of action (`actionType`) with the corresponding message in different languages (see screenshot below for version 4.3 of RP).
image
I think we need a special activity messages map for this case.
You can find some necessary messages at the `src/components/widgets/tables/projectActivity` & `..../projectActivity/activities`.
@DiscoElevator what do you think?

File: app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.jsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+ActionColumn.propTypes = {
+  className: PropTypes.string.isRequired,
+  value: PropTypes.object,
+};
+ActionColumn.defaultProps = {
+  value: {},
+};
+
+const ObjectTypeColumn = ({ className, value }) => (
+  <div className={cx('object-type-col', className)}>{value.objectType}</div>
```
Comment (AmsterGet) [inline app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.jsx]: The same is about `objectType` (see the comment above).

File: app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.jsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+);
+OldValueColumn.propTypes = {
+  className: PropTypes.string.isRequired,
+  value: PropTypes.object,
+};
+OldValueColumn.defaultProps = {
+  value: {},
+};
+
+const NewValueColumn = ({ className, value }) => (
```
Comment (AmsterGet) [inline app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.jsx]: Can you avoid duplicating this function in the [`OldValueColumn`](https://github.com/reportportal/service-ui/pull/1143/files#diff-ff6769c5845d825e3ac5407c1bba3906R78)? They are definitely similar. F.e. put some `customProps` into the column to define which type is it (`newValue` or `oldValue` key).

File: app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.jsx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    },
+    {
+      id: 'action',
+      title: {
+        full: this.props.intl.formatMessage(messages.actionCol),
+      },
+      component: ActionColumn,
+    },
+    {
+      id: 'object_type',
```
Comment (AmsterGet) [inline app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.jsx]: Please try to use camelCase everywhere.

File: app/src/pages/admin/projectEventsPage/projectEventsPage.jsx [hunk 7]
```diff
@@ -58,6 +60,7 @@ export class ProjectEventsPage extends Component {
     onChangePageSize: () => {},
     fetchEventsAction: () => {},
```
Comment (AmsterGet) [inline app/src/pages/admin/projectEventsPage/projectEventsPage.jsx]: Do we still need this action?

File: app/src/pages/admin/projectEventsPage/projectEventsPage.jsx [hunk 8]
```diff
@@ -68,9 +71,11 @@ export class ProjectEventsPage extends Component {
...
       onChangePageSize,
       loading,
+      events,
     } = this.props;
     return (
       <React.Fragment>
+        <EventsGrid data={events} loading={loading} />
         {!!pageCount &&
           !loading && (
             <PaginationToolbar
```
Comment (AmsterGet) [inline app/src/pages/admin/projectEventsPage/projectEventsPage.jsx]: Please, also fix the pagination issues (see the screenshot below).
image

Comment (Shyscarecrow) [reply app/src/pages/admin/projectEventsPage/projectEventsPage.jsx]: We have a separate task for this bug, so I'll try to fix it there

File: app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.jsx (hunk 3 above)
Comment (DiscoElevator) [reply app/src/pages/admin/projectEventsPage/eventsGrid/eventsGrid.jsx]: Sounds good!

**Pull request merged**